doc/classes/EditorTranslationParserPlugin.xml

```diff
@@ -106,14 +107,68 @@
 		<method name="_get_recognized_extensions" qualifiers="virtual const">
 			<return type="PackedStringArray" />
 			<description>
-				Gets the list of file extensions to associate with this parser, e.g. [code]["csv"][/code].
+				Override this method to provide the list of file extensions to associate with this parser, e.g. [code]["csv"][/code]. See also [method get_recognized_extensions].
 			</description>
 		</method>
 		<method name="_parse_file" qualifiers="virtual">
 			<return type="PackedStringArray[]" />
 			<param index="0" name="path" type="String" />
 			<description>
-				Override this method to define a custom parsing logic to extract the translatable strings.
+				Override this method to define a custom parsing logic to extract the translatable strings. See also [method parse_file].
+			</description>
+		</method>
+		<method name="get_all_recognized_extensions" qualifiers="static">
+			<return type="PackedStringArray" />
+			<description>
+				Returns a list of all file extensions supported by any translation parser, both standard and custom.
```

Comment:
I wonder what this method is for?
If you try to use it inside `_get_recognized_extensions()` to "override all parsers", you end up with stack overflow.

Reply:
This can be useful if you want to traverse the entire resource file system and extract all translatable strings. You don't even need to implement a parser plugin, you can use a tool script and the static methods.

Unlike `get_previous_parser()`, `get_parser()` generates a warning if no parser is found for the extension, so the obvious way to avoid unwanted warnings is to first check that the parser exists using `get_all_recognized_extensions()`.

But even if we remove the warning, I think it's useful to have access to this information. This method exists in the unexposed singleton `EditorTranslationParser` and similar functionality is available in `ResourceLoader`.